tools/validators/instance_validator/validate/instance_parser.py

```diff
@@ -101,31 +124,31 @@ def parse_yaml(filename):
     Returns the parsed YAML data in a strictyaml-provided datastructure
     which is similar to a Python dictionary.
   """
-  yaml = _load_yaml_with_schema(filename, _SCHEMA)
-
-  top_name = yaml.keys()[0]
-
-  if _TRANSLATION in yaml[top_name].keys():
-    translation = yaml[top_name][_TRANSLATION]
-    translation.revalidate(_TRANSLATION_SCHEMA)
-
-    # if translation is not UDMI compliant
-    if isinstance(translation.data, str):
-      if translation.data != _COMPLIANT:
-        print('Translation compliance improperly defined')
-        sys.exit(0)
-    else:
-      translation_keys = translation.keys()
-
-      for key in translation_keys:
-        try:
-          translation[key].revalidate(_TRANSLATION_DATA_SCHEMA)
-        except (ruamel.yaml.parser.ParserError,
-                syaml.exceptions.YAMLValidationError,
-                syaml.exceptions.DuplicateKeysDisallowed,
-                syaml.exceptions.InconsistentIndentationDisallowed,
-                ruamel.yaml.scanner.ScannerError) as exception:
-          print(exception)
-          sys.exit(0)
-
-  return yaml
+  print('Loading yaml file to validate with schema')
```

Comment:
Your question from prior comment:
"did some tests, the fastest is to keep it per entity.
Do you still want it to be configurable?"

I don't have a strong opinion about configurability (I think it's good, generally, because the constant definition provides a good opportunity to add a comment on the splitting logic, and you could imagine reasons why changing the chunk size would be useful), but I do think the logic of my proposed approach is much cleaner and more readable.

Reply:
updated